actionview/lib/action_view/template/types.rb

```diff
@@ -28,30 +28,30 @@ def initialize(symbol)
           @symbol = symbol.to_sym
         end
 
-        delegate :to_s, :to_sym, :to => :symbol
+        def to_s
+          @symbol.to_s
+        end
         alias to_str to_s
 
         def ref
-          to_sym || to_s
+          @symbol
         end
+        alias to_sym ref
 
         def ==(type)
-          return false if type.blank?
-          symbol.to_sym == type.to_sym
+          @symbol == type.to_sym unless type.blank?
         end
       end
 
-      cattr_accessor :type_klass
-
-      def self.delegate_to(klass)
-        self.type_klass = klass
-      end
-
-      delegate_to Type
+      cattr_accessor(:type_klass) { Type }
 
       def self.[](type)
         type_klass[type]
       end
+
+      def symbols
+        type_klass::SET.symbols
```

Comment:
Figured it would work since Sets and Arrays are both Enumerables, but:

```irb
irb(main):012:0> formats = [ :rss ]
=> [:rss]
irb(main):013:0> formats &= SET.symbols
TypeError: no implicit conversion of Set into Array
	from (irb):13:in `&'
	from (irb):13
	from /Users/kasperhansen/.rbenv/versions/2.2.3/bin/irb:11:in `<main>'
```